src/charm.py

```diff
@@ -224,6 +189,8 @@ def _on_config_changed(self, event: EventBase):
             self.unit.status = WaitingStatus("Waiting for pfcp agent container to be ready")
             event.defer()
             return
+        del self._kubernetes_multus
+        self._kubernetes_multus = self._get_kubernetes_multus()
```

Comment:
Why do we need this? 
We'll get new instance of the charm every time Juju event is fired and we initialize the lib in `__init__` anyway.

Reply:
We have strange issue, if we configure the MTU as default in config.yaml, it's set as expected. If we use `juju config upf  access-interface-mtu-size=1200`, it has no effect. According to juju debug-log, it deletes and creates network attachments but  access-br, core-br MTU sizes are stay same. 
```
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: GET https://10.152.183.1/apis/k8s.cni.cncf.io/v1/namespaces/core/network-attachment-definitions "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: DELETE https://10.152.183.1/apis/k8s.cni.cncf.io/v1/namespaces/core/network-attachment-definitions/core-net "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log NetworkAttachmentDefinition core-net deleted
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: POST https://10.152.183.1/apis/k8s.cni.cncf.io/v1/namespaces/core/network-attachment-definitions "HTTP/1.1 201 Created"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log NetworkAttachmentDefinition core-net created
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: GET https://10.152.183.1/apis/apps/v1/namespaces/core/statefulsets/upf "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: GET https://10.152.183.1/apis/k8s.cni.cncf.io/v1/namespaces/core/network-attachment-definitions/access-net "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: GET https://10.152.183.1/apis/k8s.cni.cncf.io/v1/namespaces/core/network-attachment-definitions/core-net "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: GET https://10.152.183.1/apis/apps/v1/namespaces/core/statefulsets/upf "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log HTTP Request: GET https://10.152.183.1/api/v1/namespaces/core/pods/upf-0 "HTTP/1.1 200 OK"
unit-upf-0: 13:43:29 INFO unit.upf/0.juju-log Default core network route created
unit-upf-0: 13:43:30 INFO unit.upf/0.juju-log No `fiveg_n3` relations found.
unit-upf-0: 13:43:30 INFO juju.worker.uniter.operation ran "config-changed" hook (via hook dispatching script: dispatch)
```

Do you want to check this issue together ?

Reply:
What I am seeing is that even though the NADs are deleted, the ifaces remain on the host until reboot.  I suspect there is something missing in the cleanup

Reply:
I saw the same behaviour, although there is no NAD, bridges in the host machines stays there. This is also happened by using kubectl with some sample pods.

Reply:
It looks like manual intervention is required: in order to change the MTU, all applications using the same network must be removed.  Next remove the bridge using `ip link down core-br` or whatever the name is.  You can then redeploy the app with a new MTU and it will take effect